server/graphql/general/src/queries/sync_status.rs

```diff
@@ -0,0 +1,96 @@
+pub use async_graphql::*;
+use chrono::NaiveDateTime;
+use graphql_core::ContextExt;
+use service::sync::sync_status::{
+    get_latest_sync_status, is_initialised, number_of_records_in_push_queue,
+};
+
+#[derive(SimpleObject)]
+pub struct SyncStatusNode {
+    started: Option<NaiveDateTime>,
+    finished: Option<NaiveDateTime>,
+}
+
+#[derive(SimpleObject)]
+pub struct SyncStatusWithProgressNode {
+    started: Option<NaiveDateTime>,
+    finished: Option<NaiveDateTime>,
+    total_progress: u32,
+    done_progress: u32,
+}
+
+#[derive(SimpleObject)]
+pub struct FullSyncStatusNode {
+    is_syncing: bool,
+    error: Option<String>,
+    summary: SyncStatusNode,
+    prepare_initial: Option<SyncStatusNode>,
+    integration: Option<SyncStatusNode>,
+    pull_central: Option<SyncStatusWithProgressNode>,
+    pull_remote: Option<SyncStatusWithProgressNode>,
+    push: Option<SyncStatusWithProgressNode>,
+}
+
+#[derive(Default, Clone)]
+pub struct SyncInfoQueries;
+
+#[Object]
+impl SyncInfoQueries {
+    pub async fn is_initialised(&self, ctx: &Context<'_>) -> Result<bool> {
+        let connection = ctx.service_provider().connection()?;
+        let is_initialised = is_initialised(&connection)?;
+
+        Ok(is_initialised)
+    }
+
+    pub async fn latest_sync_status(&self, ctx: &Context<'_>) -> Result<FullSyncStatusNode> {
+        let connection = ctx.service_provider().connection()?;
+        let sync_status = get_latest_sync_status(&connection)?;
```

Comment:
ideally get_latest_sync_status should be a service, mainly because we can test the mapping between what's returned by get_latest_sync_status and resulting graphql